lib/rubygems/remote_fetcher.rb

```diff
@@ -343,31 +343,71 @@ def close_all
 
   protected
 
+  S3Config = Struct.new :access_key_id, :secret_access_key, :security_token, :region
+
   # we have our own signing code here to avoid a dependency on the aws-sdk gem
```

Comment:
:thinking: Is this comment about simplicity still valid? This looks more complex on first sight. :eyes:

Reply:
Removed this line in fb62d30.